Makefile

```diff
@@ -9,23 +9,23 @@ INCLUDE_FIPS202 = -I fips202
 INCLUDE_MLKEM = -I mlkem
 INCLUDE_RANDOM = -I randombytes
 INCLUDE_NISTRANDOM = -I test/nistrng
-CFLAGS += -Wall -Wextra -Wpedantic -Werror -Wmissing-prototypes -Wredundant-decls \
+override CFLAGS += -Wall -Wextra -Wpedantic -Werror -Wmissing-prototypes -Wredundant-decls \
```

Comment:
What does that do?

Reply:
So we think it might be nice to be able to add some CFLAGS from command line arguments, so that things like `mcpu`, `werror`, ... could be configurable without adding a bunch of `Makefile` variables.

But it seems like we have to add this override to every `CFLAGS += ...` so that `CFLAGS` in the Makefile will not be overwritten by the `CFLAGS` from the command line arguments.

However, I don't know if there is any better solution for this.

Reply:
@potsrevennil I don't yet follow, isn't that what the `+=` would take care of? The initial value of `CFLAGS` would be amended to, not overwritten.

If it's an issue of priority, could one just make sure to append to the _left_ rather than the _right_, so command line `CFLAGS` take priority over defaults?

Reply:
According to GNU Make reference manual
> If a variable has been set with a command argument (see [Overriding Variables](https://www.gnu.org/software/make/manual/make.html#Overriding)), then ordinary assignments in the makefile are ignored. If you want to set the variable in the makefile even though it was set with a command argument, you can use an override directive,

Unfortunately, only `+=` won't be able to take care of this.

So I'm afraid that this is the only way that we can do except for setting a bunch of `Makefile` variables, if we want to allow user for more flexibility (setting `CFLAGS` from cli).

Reply:
@potsrevennil This may depend on how the command line arguments are specified?

With this dummy Makefile
```
TEST ?=
TEST += beef

all:
	@echo $(TEST)
```
I get:
```
beckphan@884d7ce1daa2 ~ % make -f test_makefile
beef
beckphan@884d7ce1daa2 ~ % TEST=dead make -f test_makefile
dead beef
beckphan@884d7ce1daa2 ~ % make TEST=dead -f test_makefile
dead
```

Reply:
Oh yes, you're right.
I assumed that we wanted the command line arguments specified as `make TEST=dead ...`.
If that is not the case, then we can indeed remove override directive